### ISSUE TITLE
Registry server: should startup require a working connection to a database?

### ISSUE THREAD

**Issue by timburks** [apigee/registry#723]
The server currently exits with an error if it is unable to make a database connection. I've seen this cause problems in two situations so far, both associated with running `registry-server` on Cloud Run:
1. When Cloud Run is autoscaling in a high-load situation, sometimes it scales up when the database is out of connections. The attempt to connect fails as a result, and the instance fails to start.
2. When the server is initially deployed to Cloud Run, the deployment can fail because the database connection isn't fully enabled. Here's a sample error message:
> Failed to create registry server error rpc error: code = Internal desc = ensure that the account has access to "XXXXX:us-west1:registry" (and make sure there's no typo in that name). Error during generateEphemeral for XXXXX:us-west1:registry: googleapi: Error 403: Cloud SQL Admin API has not been used in project XXXXX before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=XXXXX then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
Enabling the Cloud SQL Admin API can take several minutes to fully propagate. Even though it shows as enabled in the console, it still might not be available to the Cloud Run servers.

Are there other situations where the server fails to start due to system configuration? It might be better to let it start and let requests fail until the database becomes available.

**Comment by theganyo** [comment]
I agree, we should ensure the server will connect and reconnect to the db appropriately, then let the server start but fail all requests with a 503 Service Unavailable error - and return the same if the db becomes unavailable later.
